buildpipe/pipeline.py

```diff
@@ -101,68 +98,68 @@ def _update_dicts(source: Dict, overrides: Dict) -> Dict:
 
 def buildkite_override(step_func: Callable) -> Callable:
     @functools.wraps(step_func)
-    def func_wrapper(stair: box.Box, projects: Set[box.Box]) -> List[Dict]:
-        return [_update_dicts(step, stair.buildkite.to_dict()) for step in step_func(stair, projects)]
+    def func_wrapper(stair: Dict, projects: List[Dict]) -> List[Dict]:
+        return [_update_dicts(step, stair.get('buildkite', {})) for step in step_func(stair, projects)]
     return func_wrapper
 
 
 def generate_default_wait_step() -> List[str]:
     return ['wait']
 
 
-def generate_wait_step(stair: box.Box) -> List[str]:
-    if stair.continue_on_failure:
+def generate_wait_step(stair: Dict) -> List[str]:
+    if stair.get('continue_on_failure'):
         return [{'wait': None, 'continue_on_failure': True}]
     else:
         return generate_default_wait_step()
 
 
-def generate_block_step(block: Dict, stair: box.Box, projects: Set[box.Box]) -> List[Dict]:
-    has_block = any(stair.name in _get_block(project) for project in projects)
+def generate_block_step(block: Dict, stair: Dict, projects: List[Dict]) -> List[Dict]:
+    has_block = any(stair['name'] in _get_block(project) for project in projects)
     return [block] if has_block else []
 
 
 @buildkite_override
-def generate_project_steps(stair: box.Box, projects: Set[box.Box]) -> List[Dict]:
+def generate_project_steps(stair: Dict, projects: List[Dict]) -> List[Dict]:
     steps = []
     for project in projects:
         step = {
-            'label': f'{stair.name} {project.name} {stair.emoji or project.emoji or ""}'.strip(),
+            'label': f'{stair["name"]} {project["name"]} {stair.get("emoji") or project.get("emoji") or ""}'.strip(),
             'env': {
-                'BUILDPIPE_STAIR_NAME': stair.name,
-                'BUILDPIPE_STAIR_SCOPE': stair.scope,
-                'BUILDPIPE_PROJECT_NAME': project.name,
-                'BUILDPIPE_PROJECT_PATH': project.path,
+                'BUILDPIPE_STAIR_NAME': stair['name'],
+                'BUILDPIPE_STAIR_SCOPE': stair['scope'],
+                'BUILDPIPE_PROJECT_NAME': project['name'],
+                'BUILDPIPE_PROJECT_PATH': project['path'],
                 # Deprecated environment variable names
                 # TODO: remove when cutover to new minor version
-                'STAIR_NAME': stair.name,
-                'STAIR_SCOPE': stair.scope,
-                'PROJECT_NAME': project.name,
-                'PROJECT_PATH': project.path,
+                'STAIR_NAME': stair['name'],
+                'STAIR_SCOPE': stair['scope'],
+                'PROJECT_NAME': project['name'],
+                'PROJECT_PATH': project['path'],
                 # Add other environment variables specific to project
-                **(project.env or {})
+                **project.get('env', {})
```

Comment:
This won't do the same in case `env` is given some invalid value like `0` or `null`:
```python
>>> {'env': None}.get('env', {})
None
>>> {'env': None}['env'] or {}
{}
```
Doubt it's important, but it's technically interface-breaking.